lgc/builder/BuilderRecorder.cpp

```diff
@@ -1848,10 +1847,147 @@ Instruction *BuilderRecorder::record(BuilderRecorder::Opcode opcode, Type *resul
+    case Opcode::MatrixTimesVector:
+    case Opcode::NormalizeVector:
+    case Opcode::OuterProduct:
+    case Opcode::QuantizeToFp16:
+    case Opcode::Reflect:
+    case Opcode::Refract:
+    case Opcode::SAbs:
+    case Opcode::SMod:
+    case Opcode::SSign:
+    case Opcode::SmoothStep:
+    case Opcode::TransposeMatrix:
+    case Opcode::VectorTimesMatrix:
+    case Power:
+    case Sinh:
+    case Tan:
+    case Tanh:
+      // Functions that don't access memory.
+      func->addFnAttr(Attribute::ReadNone);
+      break;
+    case Opcode::ImageGather:
+    case Opcode::ImageLoad:
+    case Opcode::ImageLoadWithFmask:
+    case Opcode::ImageSample:
+    case Opcode::LoadDescFromPtr:
+    case Opcode::LoadPushConstantsPtr:
+    case Opcode::ReadBuiltInInput:
+    case Opcode::ReadBuiltInOutput:
+    case Opcode::ReadGenericInput:
+    case Opcode::ReadGenericOutput:
+      // Functions that only read memory.
+      func->addFnAttr(Attribute::ReadOnly);
+      break;
+    case Opcode::ImageStore:
+      // Functions that only write memory.
+      func->addFnAttr(Attribute::WriteOnly);
+      break;
+    case Opcode::ImageAtomic:
+    case Opcode::ImageAtomicCompareSwap:
+      // Functions that read and write memory.
+      break;
+    case Opcode::Barrier:
+    case Opcode::DemoteToHelperInvocation:
+    case Opcode::EmitVertex:
+    case Opcode::EndPrimitive:
+    case Opcode::GetSubgroupSize:
+    case Opcode::ImageGetLod:
+    case Opcode::ImageQueryLevels:
+    case Opcode::ImageQuerySamples:
+    case Opcode::ImageQuerySize:
+    case Opcode::IsHelperInvocation:
+    case Opcode::Kill:
+    case Opcode::LoadBufferDesc:
+    case Opcode::ReadClock:
+    case Opcode::SubgroupAll:
+    case Opcode::SubgroupAllEqual:
+    case Opcode::SubgroupAny:
+    case Opcode::SubgroupBallot:
+    case Opcode::SubgroupBallotBitCount:
+    case Opcode::SubgroupBallotBitExtract:
+    case Opcode::SubgroupBallotExclusiveBitCount:
+    case Opcode::SubgroupBallotFindLsb:
+    case Opcode::SubgroupBallotFindMsb:
+    case Opcode::SubgroupBallotInclusiveBitCount:
+    case Opcode::SubgroupBroadcast:
+    case Opcode::SubgroupBroadcastFirst:
+    case Opcode::SubgroupClusteredExclusive:
+    case Opcode::SubgroupClusteredInclusive:
+    case Opcode::SubgroupClusteredReduction:
+    case Opcode::SubgroupElect:
+    case Opcode::SubgroupInverseBallot:
+    case Opcode::SubgroupMbcnt:
+    case Opcode::SubgroupQuadBroadcast:
+    case Opcode::SubgroupQuadSwapDiagonal:
+    case Opcode::SubgroupQuadSwapHorizontal:
+    case Opcode::SubgroupQuadSwapVertical:
+    case Opcode::SubgroupShuffle:
+    case Opcode::SubgroupShuffleDown:
+    case Opcode::SubgroupShuffleUp:
+    case Opcode::SubgroupShuffleXor:
+    case Opcode::SubgroupSwizzleMask:
+    case Opcode::SubgroupSwizzleQuad:
+    case Opcode::SubgroupWriteInvocation:
+    case Opcode::WriteBuiltInOutput:
+    case Opcode::WriteGenericOutput:
+    case Opcode::WriteXfbOutput:
```

Comment:
Transform feedback output will read the stream out buffer descriptor and make a tbuffer store. So it could be marked as readwrite

Reply:
#775